docs/tutorials/mongoid-transactions.txt

```diff
@@ -0,0 +1,63 @@
+=========
+Sessions
+=========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Versions 4.0 and higher of the MongoDB server support transactions. You can use sessions with Mongoid in a similar way
+that you would execute a transaction in ActiveRecord as long as the current thread is within a session. Namely, you can
+call a method, ``#with_transaction`` on a model class or on an instance of a model from within a session block and execute
+some operations in a new block. All operations in the inner block will be executed in the context of single transactions. Please
+see the MongoDB Ruby driver documentation for what transaction options are available.
+
+Please note the following limitations of sessions:
+
+- Transactions cannot be shared across threads; transactions are not thread-safe. This is consistent with the Ruby driver's support for transactions.
+
+- Transactions cannot be nested. You cannot called ``#with_transaction`` on a model class or a model instance within the block passed to the ``#with_transaction`` method on another model class or model instance.
```

Comment:
It is my impression that ActiveRecord allows nesting transactions, and if so, Mongoid should too.

Reply:
I don't think that's possible; MongoDB only supports one open transaction at a time on a given session